### PR TITLE
Internal: Remove `Atomic_Prop` class [EDS-408]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 1]
```diff
@@ -136,33 +132,6 @@ public static function get_props_schema(): array {
...
-
-			if ( ! $prop_type ) {
-				Utils::safe_throw( "Prop type `{$prop->get_type()}` for prop `$key` does not exist in `{$widget_name}`." );
-			}
-
-			try {
-				$prop->validate( $prop->get_default() );
-			} catch ( \Exception $e ) {
-				Utils::safe_throw( "Default value for `$key` prop is invalid in `{$widget_name}` - {$e->getMessage()}" );
-			}
```
Comment (StyleShit) [inline modules/atomic-widgets/base/atomic-widget-base.php:162]: @Nevoss 
do you think we should keep this?

File: tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php [hunk 2]
```diff
@@ -203,37 +195,39 @@ public function test_get_props_schema__is_serializable() {
...
+				]
 			},
 			"number_prop": {
 				"type": "number",
-				"constraints": [],
-				"default": 123
+				"default": 123,
+				"dynamic_categories": [
+					"number"
+				]
```
Comment (StyleShit) [inline tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php:209]: @Nevoss 

I don't like the fact that we check the "dynamic_categories" here
any idea how tp avoid that?

File: modules/atomic-widgets/prop-types/prop-type.php [hunk 3]
```diff
@@ -20,9 +42,10 @@ public function get_dynamic_categories(): array {
...
 
 	public function jsonSerialize(): array {
-		return [
-			'type' => $this->get_type(),
+		return array_merge( [
+			'key' => $this->get_key(),
 			'dynamic_categories' => $this->get_dynamic_categories(),
-		];
+			'default' => $this->default,
+		], $this->settings );
```
Comment (StyleShit) [inline modules/atomic-widgets/prop-types/prop-type.php:49]: the dynamic categories will now be passed to the editor on each prop in the schema, which means there will be duplicates. Do we care about it?

File: modules/atomic-widgets/base/atomic-widget-base.php (hunk 1 above)
Comment (Nevoss) [reply modules/atomic-widgets/base/atomic-widget-base.php:162]: We should still validate the schema but without validating the default value, maybe?

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 4]
```diff
@@ -26,9 +25,6 @@ public function get_atomic_controls() {
 		$controls = $this->define_atomic_controls();
 		$schema = static::get_props_schema();
 
-		// Validate the schema only in the Editor.
-		static::validate_schema( $schema );
```
Comment (Nevoss) [inline modules/atomic-widgets/base/atomic-widget-base.php:30]: I think we can keep it, but maybe make it simpler

File: modules/atomic-widgets/prop-types/prop-type.php [hunk 5]
```diff
@@ -8,10 +8,32 @@
 
 abstract class Prop_Type implements \JsonSerializable {
 
-	abstract public function get_type(): string;
+	protected $default = null;
+
+	// TODO: Find a real name!
```
Comment (Nevoss) [inline modules/atomic-widgets/prop-types/prop-type.php:13]: It is fine, remove 😄

File: modules/atomic-widgets/prop-types/prop-type.php (hunk 3 above)
Comment (Nevoss) [reply modules/atomic-widgets/prop-types/prop-type.php:49]: Can we do it differently? just pass some configuration on the "dynamic module" (we do not have a module for dynamic but we can create a class that is responsible for it) also the category should not be hardcoded in the prop type

File: modules/atomic-widgets/prop-types/string-prop-type.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	/**
+	 * @return array<string>
+	 */
+	public function get_dynamic_categories(): array {
+		return [
+			DynamicTags::TEXT_CATEGORY,
+		];
+	}
```
Comment (Nevoss) [inline modules/atomic-widgets/prop-types/string-prop-type.php:45]: Remove

File: tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php (hunk 2 above)
Comment (Nevoss) [reply tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php:209]: I feel it should live outside of the `Prop_Types` mechanism and be independent

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/base/atomic-widget-base.php (hunk 1 above)
Comment (StyleShit) [reply modules/atomic-widgets/base/atomic-widget-base.php:162]: I think validating the default is fine
added the validation back, maybe we'll remove it in the future

File: modules/atomic-widgets/prop-types/prop-type.php (hunk 5 above)
Comment (StyleShit) [reply modules/atomic-widgets/prop-types/prop-type.php:13]: Dםמק

File: modules/atomic-widgets/prop-types/prop-type.php [hunk 7]
```diff
@@ -8,21 +8,35 @@
...
 
 	public function jsonSerialize(): array {
-		return [
-			'type' => $this->get_type(),
-			'dynamic_categories' => $this->get_dynamic_categories(),
-		];
+		return array_merge( [
+			'type' => static::get_key(),
+			'default' => $this->default,
+		], $this->settings );
```
Comment (StyleShit) [inline modules/atomic-widgets/prop-types/prop-type.php:40]: put under settings key

File: modules/atomic-widgets/dynamic-tags.php [hunk 8]
```diff
@@ -5,17 +5,22 @@
...
 	public function register_hooks() {
 		add_filter( 'elementor/editor/localize_settings', fn( $settings ) => $this->add_atomic_dynamic_tags_settings( $settings ) );
 	}
 
 	private function add_atomic_dynamic_tags_settings( $settings ) {
 		if ( isset( $settings['dynamicTags']['tags'] ) ) {
-			$settings['atomicDynamicTags'] = $this->convert_dynamic_tags_to_atomic( $settings['dynamicTags']['tags'] );
+			$settings['atomicDynamicTags'] = [
+				'tags' => $this->convert_dynamic_tags_to_atomic( $settings['dynamicTags']['tags'] ),
+				'prop_types_mapping' => $this->get_prop_types_mapping(),
```
Comment (StyleShit) [inline modules/atomic-widgets/dynamic-tags.php:22]:
```suggestion
				'prop_types_to_dynamic' => $this->get_prop_types_mapping(),
```

**Review by Nevoss — APPROVED** [review]

**Review by Nevoss — APPROVED** [review]

**Pull request merged**